AWS.hs

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env nix-shell
+#! nix-shell -j 4 -i runhaskell -p 'pkgs.haskellPackages.ghcWithPackages (hp: with hp; [ turtle ])'
+#! nix-shell -I nixpkgs=https://github.com/NixOS/nixpkgs/archive/f2c4af4e3bd9ecd4b1cf494270eae5fd3ca9e68c.tar.gz
+
+{-# LANGUAGE OverloadedStrings, UnicodeSyntax #-}
+
+import Control.Monad (forM_)
+import Data.Maybe
+import Data.Monoid ((<>))
+import Filesystem.Path.CurrentOS (encodeString)
+import Prelude hiding (FilePath)
+import qualified Data.Text as T
+import qualified Data.Text.IO as TIO
+import Turtle
+
+
+-- * URLs
+data URL =
+  URL
+  { proto :: T.Text
+  , host  :: T.Text
+  , key   :: T.Text
+  } deriving (Show)
+
+ppURL:: URL -> T.Text
+ppURL (URL pro ho ke) = pro <> "://" <> ho <> "/" <> ke
+
+
+-- * Buckets
+data Bucket     = Bucket { fromBucket ∷ T.Text } deriving (Show)
```

Comment:
Why the oddball unicode symbol `∷`? I notice the `UnicodeSyntax` extension is enabled too, but for the most part you're not using it.

My personal preference is avoiding features like this unless we make a style guide that explicitly encourages it. Consistency is good.